### PR TITLE
docs: add more context to the strict mode guide

### PR DESCRIPTION
**Pull request by mgechev**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/content/guide/strict-mode.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Strict mode
 
+Stricter type checking and other similar constraints often enable more optimizations and correctness verification at build time. You can learn more about the theory behind this [here](https://www.youtube.com/watch?v=X-Dn5ZBUZH0). At the same time, constraints may hurt development experience (DX).
```
Comment (IgorMinar) [inline aio/content/guide/strict-mode.md:3]: The last sentence sounds a bit scary. Maybe if we worded it slightly differently, we could explain the value of strictness better and encourage developers into making the trade offs - most of which are short term trade offs and will pay for themselves quickly.

File: aio/content/guide/strict-mode.md [hunk 2]
```diff
@@ -1,5 +1,11 @@
...
+
+In Angular, we want to enable the strictest flags to empower best practices, without significantly impacting the learning curve when getting started with the framework, nor DX.
```
Comment (IgorMinar) [inline aio/content/guide/strict-mode.md:5]:
```suggestion
In Angular, we want to enable the strictest flags to encourage best practices, without significantly impacting the learning curve when getting started with the framework, nor DX.
```

**Review by IgorMinar — APPROVED** [review]
From the technical point of view this looks good to me with a few nits, but I'd prefer for the change to be more concise. I'll leave it up to @aikidave to review for that and recommend modifications.

File: aio/content/guide/strict-mode.md [hunk 3]
```diff
@@ -1,5 +1,11 @@
...
+
+As an experiment that will allow us to understand the limits we can push, we implemented strict mode. We are using the strict mode to encourage the adoption of best practices and understand the strict options developers will feel comfortable with while still enabling most compile-time correctness checks and optimizations. You can learn more about the strict flags we've been experimenting with and their implications [here](https://blog.angular.io/angular-cli-strict-mode-c94ba5965f63).
+
 When you create a new workspace or an application you have an option to create them in a strict mode using the `--strict` flag.
 
 Enabling this flag initializes your new workspace or application with a few new settings that improve maintainability, help you catch bugs ahead of time.
```
Comment (alan-agius4) [inline aio/content/guide/strict-mode.md:11]: Can we also add that this will improve the DX when using LS?

Maybe @kyliau can suggest the wording.

Comment (mgechev) [reply aio/content/guide/strict-mode.md:11]: I'd actually suggest holding off on this. We can confuse people between the two different versions of the language service. We should definitely increase the visibility of this once the Ivy LS becomes stable and the recommended solution.

**Review by alan-agius4 — APPROVED** [review]
LGTM, thanks

File: aio/content/guide/strict-mode.md [hunk 4]
```diff
@@ -1,5 +1,11 @@
 # Strict mode
 
+Stricter type checking and other similar constraints often enable more optimizations and correctness verification at build time. You can learn more about the theory behind this [here](https://www.youtube.com/watch?v=X-Dn5ZBUZH0). At the same time, constraints may have an impact on development experience (DX).
```
Comment (aikidave) [inline aio/content/guide/strict-mode.md:3]:
```suggestion
Stricter type checking and other similar constraints often enable more optimizations and correctness verification at build time. For more information about this theory, see [Power in Constraints](https://www.youtube.com/watch?v=X-Dn5ZBUZH0). At the same time, constraints may impact the development experience.
```
Please avoid using "here" as link text. It's not very accessible for those using screen readers, and so on.

File: aio/content/guide/strict-mode.md [hunk 5]
```diff
@@ -1,5 +1,11 @@
...
+
+In Angular, we want to enable the strictest flags to encourage best practices, without significantly impacting the learning curve when getting started with the framework, nor DX.
+
+As an experiment that will allow us to understand the limits we can push, we implemented strict mode. We are using the strict mode to encourage the adoption of best practices and understand the strict options developers will feel comfortable with while still enabling most compile-time correctness checks and optimizations. You can learn more about the strict flags we've been experimenting with and their implications [here](https://blog.angular.io/angular-cli-strict-mode-c94ba5965f63).
```
Comment (aikidave) [inline aio/content/guide/strict-mode.md:7]: This passage reads like it belongs in a blog post; not in the documentation. Remember, the goal of the documentation is to help users get things done. In addition, this passage doesn't lead into the next sentence, which actually starts the describing how to use strict mode.

I do understand that we should explain what strict mode is and why developers should use it; however, I don't think these two paragraphs belong in the documentation.

**Pull request closed without merging**